UPGRADE.md

```diff
@@ -10,6 +10,8 @@ UPGRADE
 * Removed specific ``ZoneMembers`` i.e. ``ProvinceZoneMember`` in favor of a dynamic ``ZoneMember``;
 * https://github.com/Sylius/Sylius/pull/3696
 
+* Replaced ``Province`` with ``AdministrativeArea``
```

Comment:
this is a very big BC break on addressing and I think it needs a lot more comments here.